### PR TITLE
table link color fix

### LINKED ISSUE #12766: light theme uses dark color for panel links with row colors
### What Grafana version are you using?
5.3-pre1
### What datasource are you using?
n/a
### What OS are you running grafana on?
Debian 9
### What did you do?
- switched theme to light
- within a table panel:
  - add column style and check "render value as link"
  - add column style with a row color
### What was the expected result?
normal text and links should use white color
### What happened instead?
text turns white, links stay dark

image

### PR DESCRIPTION
**Pull request by Ijin08**
fixed color for links in colored cells by adding a new variable that sets color: white when cell or row has background-color

fixes #12766

### CODE REVIEW

**Review by marefr — COMMENTED** [review]
Would be convenient to use css classes instead of style. Have you considered that? Basically adding a class if row/cell/value color is active which can be used to style the links.

Render column as link plus threshold with color mode value still render link as white/dark. But I see that this is an existing problem so lets skip this fix for now.

It's quite hard to spot that it actually are links in a table, but that's probably a bigger problem not solved now.

**Comment by Ijin08** [comment]
@marefr I will add a class instead and maybe in the future it can be made more clear that they are links.

**Review by marefr — COMMENTED** [review]
That works. But wouldn't be easier to use the existing cellClasses and push a class for each color state? Then you can use css to style both color state font and link cases.

File: public/app/plugins/panel/table/renderer.ts [hunk 1]
```diff
@@ -217,12 +217,12 @@ export class TableRenderer {
 
     if (this.colorState.cell) {
       style = ' style="background-color:' + this.colorState.cell + ';color: white"';
+      cellClasses.push('white');
```
Comment (marefr) [inline public/app/plugins/panel/table/renderer.ts]: class name white is not so good naming. 

Was thinking that class names could be convenient to use in general for table cells and not only for links, but that's maybe not a good idea - what do you think?

Was thinking of something like this
* table-panel-cell-color-cell
* table-panel-cell-color-row
* table-panel-cell-color-value

Then the `style: color: white` can be removed as well and instead be styled from sass. Exception is when table-panel-cell-color-value

**Comment by Ijin08** [comment]
@marefr added classes as suggested but for value I think we must use inline style since the color is set in the render file.

**Review by marefr — APPROVED** [review]
Excellent - looks good :+1:

**Pull request merged**